docs/grafana.rst

```diff
@@ -0,0 +1,320 @@
+
+There is a configuration entry ``beam.sim.metric.collector.metrics`` which contains names of enabled metrics
+and thus controls which metrics will be written during BEAM execution.
+
+By default metrics which do not affect performance are enabled globally in “test/input/common/metrics.conf” file.
+
+To configure metrics for specific BEAM configuration one may add the configuration entry in that specific configuration file.
+
+**Metrics description:**
+
+Metrics which contain a run name and an iteration number, **necessary for displaying any metric**::
+
+    beam-run,
+    beam-iteration
+
+Metrics which contain a single number with a count of households, population size, charging stalls count e.t.c::
+
+    beam-run-households,
+    beam-run-population-size,
+    beam-run-private-fleet-size,
+    beam-run-charging-depots-cnt,
+    beam-run-charging-depots-stalls-cnt,
+    beam-run-public-fast-charge-cnt,
+    beam-run-public-fast-charge-stalls-cnt
+
+Metrics which contain a single number with a count of different types of ride hail vehicles::
+
+    beam-run-RH-ev-cav,
+    beam-run-RH-non-ev-cav,
+    beam-run-RH-ev-non-cav,
+    beam-run-RH-non-ev-non-cav
+
+Ride hail EV (electric vehicle) CAV (connected and automated vehicle) metrics::
+
+    rh-ev-cav-count,
+    rh-ev-cav-distance,
+    rh-ev-nocav-count,
+    rh-ev-nocav-distance,
+    rh-noev-cav-count,
+    rh-noev-cav-distance,
+    rh-noev-nocav-count,
+    rh-noev-nocav-distance
+
+Various metrics for all vehicles/persons::
+
+    parking,
+    chargingPower,
+    mode-choices,
+    average-travel-time
+
+Various metrics for ride hail
+
+.. code-block:: text
+
+    ride-hail-waiting-time,
+    ride-hail-waiting-time-map,
+    ride-hail-trip-distance,
+    ride-hail-inquiry-served (graph is not added to a grafana dashboard),
+    ride-hail-allocation-reserved (graph is not added to a grafana dashboard)
+
+New metrics
+^^^^^^^^^^^^^^^^
+
+In order to write and display a new metric one should do two things:
+
+* write metric into metric storage
+* display metric on a grafana dashboard.
+
+**How to write a new metric into storage**
+
+To write a new metric into storage during BEAM execution one needs to use an appropriate method from trait ``SimulationMetricCollector``:
+there are methods to write iteration-level metrics with hours on X axis or to be displayed on a map
+and methods to write global-level metrics with iteration on X axis. There is also a method to check if a metric is enabled.
+
+
+Troubleshooting
+--------------------
+
+.. error ::
+
+    for docker-influxdb-grafana Cannot create container for service docker-influxdb-grafana: Conflict. The container
+    name "/docker-influxdb-grafana" is already in use by container "<CONTAINER ID>”. You have to remove (or rename) that container to be able to reuse that name.
+
+This error means that one already has a container with name ‘docker-influxdb-grafana’ in docker.
+To coupe with that one may remove that container::
```

Comment:
To handle that...?

'm not sure what coupe was supposed to be. It seems to be a typo

Reply:
let's assume it was a typo (=